client/my-sites/plugins/plugin-management-v2/hooks/use-plugin-version-info.ts

```diff
@@ -0,0 +1,82 @@
+import { useMemo } from 'react';
+import { siteObjectsToSiteIds } from 'calypso/my-sites/plugins/utils';
+import { useSelector } from 'calypso/state';
+import { getPluginOnSites } from 'calypso/state/plugins/installed/selectors';
+import getSites from 'calypso/state/selectors/get-sites';
+import type { PluginComponentProps } from '../types';
+import type { SiteDetails } from '@automattic/data-stores';
+
+export default function usePluginVersionInfo(
+	plugin: PluginComponentProps,
+	selectedSite?: SiteDetails
+): {
+	currentVersionsRange: { min: string; max: string };
+	updatedVersions: string[];
+	hasUpdate: boolean;
+} {
+	const allSites = useSelector( getSites );
+
+	const sites = plugin?.sites
+		? Object.keys( plugin.sites ).map( ( siteId ) => {
+				const site = allSites.find( ( s ) => s?.ID === parseInt( siteId ) );
+				return {
+					...site,
+					...plugin.sites[ siteId ],
+				} as any; // This must be cast as any until this file is updated to work with the selectors in state/plugins/installed/selectors
+		  } )
+		: [];
+
+	const siteIds = siteObjectsToSiteIds( sites );
+	const state = useSelector( ( state ) => state );
```

Comment:
⚠️ By selecting the entire state (as also seen in `UpdatePlugin`), we're guaranteeing that anywhere this hook is referenced will re-render any time any aspect of the app state is changed. This could significantly impact performance. As an alternative, I recommend individual calls to `useSelector` as needed for each part of the state we need.

Reply:
Good catch. Yeah, I didn't realize this earlier. I have made the changes: https://github.com/Automattic/wp-calypso/pull/81712/commits/cc6a3a7ec3d1a8a0ae67b8c5ff08634995b27da6